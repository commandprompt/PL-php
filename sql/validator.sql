--
-- Test the validator
--

CREATE FUNCTION invalid() RETURNS void LANGUAGE plphp AS $$
	asdasd
$$;

CREATE FUNCTION invalid() RETURNS void LANGUAGE plphp AS $$
	{
$$;

CREATE OR REPLACE FUNCTION valid() RETURNS void LANGUAGE plphp AS $$
	array_append();
$$;
select valid();

CREATE FUNCTION valid2() RETURNS void LANGUAGE plphp AS $$
	return array();
$$;
select valid2();
