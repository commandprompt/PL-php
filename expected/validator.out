--
-- Test the validator
--
CREATE FUNCTION invalid() RETURNS void LANGUAGE plphp AS $$
	asdasd
$$;
ERROR:  function "invalid" does not validate: syntax error, unexpected '}' at line 3
CREATE FUNCTION invalid() RETURNS void LANGUAGE plphp AS $$
	{
$$;
ERROR:  function "invalid" does not validate: syntax error, unexpected $end at line 3
CREATE OR REPLACE FUNCTION valid() RETURNS void LANGUAGE plphp AS $$
	array_append();
$$;
select valid();
ERROR:  Call to undefined function array_append() at line 2
CREATE FUNCTION valid2() RETURNS void LANGUAGE plphp AS $$
	return array();
$$;
select valid2();
ERROR:  this plphp function cannot return arrays
